packages/codegen/__tests__/codegen.test.ts

```diff
@@ -176,44 +198,188 @@ describe('codegen', () => {
+    `);
+  });
+
+  test('gql tags with shared fragments transitive', async () => {
+    project = await createFixtures(gqlFilesMapWithSharedFragmentsTransitive);
+
+    await athenaCodegen({
+      schemaPath: 'schema.graphql',
+      documents: [`${project.baseDir}/**/*.tsx`],
+      baseDir: project.baseDir,
+      extension: '.graphql.ts',
+      production: false,
+    });
+
+    expect(await read('schema.graphql.ts')).toMatchSnapshot();
+
+    expect(await read('__generated/MessageView.graphql.ts'))
+      .toMatchInlineSnapshot(`
+      "import type * as Types from '../schema.graphql.js';
+
+      import type { TypedDocumentNode as DocumentNode } from '@graphql-typed-document-node/core';
+      export type MessageFieldsFragment = { __typename: 'ChatMessage', id: string, content: string, time: any };
+
+      export const MessageFieldsFragmentDoc = {\\"kind\\":\\"Document\\",\\"definitions\\":[{\\"kind\\":\\"FragmentDefinition\\",\\"name\\":{\\"kind\\":\\"Name\\",\\"value\\":\\"MessageFields\\"},\\"typeCondition\\":{\\"kind\\":\\"NamedType\\",\\"name\\":{\\"kind\\":\\"Name\\",\\"value\\":\\"ChatMessage\\"}},\\"selectionSet\\":{\\"kind\\":\\"SelectionSet\\",\\"selections\\":[{\\"kind\\":\\"Field\\",\\"name\\":{\\"kind\\":\\"Name\\",\\"value\\":\\"id\\"}},{\\"kind\\":\\"Field\\",\\"name\\":{\\"kind\\":\\"Name\\",\\"value\\":\\"content\\"}},{\\"kind\\":\\"Field\\",\\"name\\":{\\"kind\\":\\"Name\\",\\"value\\":\\"time\\"}}]}}]} as unknown as DocumentNode<MessageFieldsFragment, unknown>;"
+    `);
+
     expect(
-      await fs.readFile(
-        path.join(project.baseDir, 'schema.graphql.ts'),
-        'utf-8'
-      )
+      await read('__generated/ChatView.graphql.ts')
     ).toMatchInlineSnapshot(`
-      "export type Maybe<T> = T | null;
-      export type InputMaybe<T> = Maybe<T>;
-      export type Exact<T extends { [key: string]: unknown }> = { [K in keyof T]: T[K] };
-      export type MakeOptional<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]?: Maybe<T[SubKey]> };
-      export type MakeMaybe<T, K extends keyof T> = Omit<T, K> & { [SubKey in K]: Maybe<T[SubKey]> };
-      /** All built-in and custom scalars, mapped to their actual values */
-      export type Scalars = {
-        ID: string;
-        String: string;
-        Boolean: boolean;
-        Int: number;
-        Float: number;
-        Date: any;
-      };
+      "import type * as Types from '../schema.graphql.js';
 
-      export enum Role {
-        User = 'USER',
-        Admin = 'ADMIN'
-      }
-      "
+      import type { TypedDocumentNode as DocumentNode } from '@graphql-typed-document-node/core';
+      export type MessageFieldsFragment = { __typename: 'ChatMessage', id: string, content: string, time: any };
```

Comment:
🐛  chatview should not export the type `MessageFieldsFragment` -- it should only export the fragment types that it defines itself.